File: Scripts/newTask/ackermann-function/java/AckermannFunction.java
import java.math.BigInteger;

public class AckermannFunction {


public static void main(String[] args) {

    for (long m = 0; m <= 4; ++m)
    {
        for (long n = 0; n < 6 - m; ++n)
        {
            Ackermann(m,n);
        }
    }
}

public static long Ackermann(long m, long n)
{
    if(m > 0)
    {
        if (n > 0)
            return Ackermann(m - 1, Ackermann(m, n - 1));
        else if (n == 0)
            return Ackermann(m - 1, 1);
    }
    else if(m == 0)
    {
        if(n >= 0)
            return n + 1;
    }

    return 1;
}


}

File: Scripts/newTask/ackermann-function/c++/ackermann-function.cpp
#include <iostream>

unsigned int ackermann(unsigned int m, unsigned int n) {
  if (!m) return n + 1;
  if (!n) return ackermann(m - 1, 1);
  return ackermann(m - 1, ackermann(m, n - 1));
}

int main() {
  for (int i = 0; i < 50; ++i) {
  for (unsigned int m = 0; m <= 4; ++m) {
    for (unsigned int n = 0; n < 6 - m; ++n) {
      //std::cout << "A(" << m << ", " << n << ") = " << ackermann(m, n) << "\n";
      ackermann(m, n);
    }
  }
}
}
